suprb/optimizer/rule/generation_operator/base.py

```diff
@@ -0,0 +1,25 @@
+from abc import ABCMeta, abstractmethod
+from suprb.rule.matching import MatchingFunction, OrderedBound
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+from suprb.utils import RandomState
+
+
+class GenerationOperator(BaseComponent, metaclass=ABCMeta):
```

Comment:
Can we somehow signify within the name that this pertains to the rulegeneration metaheuristic? (Without calling it `RuleGenerationOperator` or even worse `RuleGenerationGenerationOperator`)